src/utils/logs-buffer.ts

```diff
@@ -12,11 +12,19 @@ export class LogsBuffer {
         this.flushTimeout = null
     }
 
-    addLog(log: LogEntryPayload): void {
+    async addLog(log: LogEntryPayload): Promise<void> {
+        // drop into kafka queue immediately as that's aready a buffer
+        if (this.db.kafkaProducer) {
+            await this.db.createPluginLogEntries([log])
+            return
+        }
```

Comment:
This is kind of confusing. So this way in Kafka/CH mode, we have a `LogsBuffer`, which goes through all the motions as usual… but it's illusory because no data is buffered in it, as the _actual_ buffering is handed of to the Kafka-specific buffer immediately.